code/game/objects/cleaning.dm

```diff
@@ -29,11 +29,12 @@
 	if(!do_after(user, cleanspeed, target = src))
 		return FALSE
 
-	cleaner.post_clean(src, user)
-
 	if(!cleaner.can_clean())
+		cleaner.post_clean(src, user)
```

Comment:
why call post clean here?

Reply:
Post clean consumes the reagent and applies it on the floor, this call is necessary for cleaning devices that dont have water/cleaner reagents in them. 
Originally, because you had to return earlier for reagents that cannot clean, the post_clean was being called before the can_clean which would lead the mop to remove their last 1 reagent before the can_clean check (if cleaning reagent < 1), that would make it return before reaching the cleaning part of the code.